services/util/parseWikiUrl.js

```diff
@@ -21,16 +23,34 @@ function extractName(url) {
   return checks.find(name => !!name);
 }
 
-function splitUrl(url) {
-  if (!regExp.test(url)) {
-    return null;
-  }
-  const matches = url.match(regExp);
+function splitUploadUrl(url) {
+  const matches = url.match(uploadRegExp);
+  const domain = matches[1] === 'commons' ? 'wikimedia' : 'wikipedia';
+  const wikiUrl = `https://${matches[1]}.${domain}.org/`;
+  const segments = url.split('/');
+  const fileName = segments.includes('thumb') ? segments[segments.length - 2] : segments.pop();
+  const title = `File:${fileName}`;
+  return { title, wikiUrl };
+}
+
+function splitWikipediaUrl(url) {
+  const matches = url.match(wikipediaRegExp);
   const wikiUrl = `https://${matches[1]}.wikipedia.org/`;
   const title = extractName(url);
   return { title, wikiUrl };
 }
 
+function splitUrl(url) {
+  if (uploadRegExp.test(url)) {
```

Comment:
Tiny idea… This could potentially be extracted into a function:  `isUploadUrl(url)`.